src/SourceLocator/Type/AutoloadSourceLocator.php

```diff
@@ -229,6 +230,10 @@ private function locateConstantByName(string $constantName) : ?string
         //       defined a constant that is being looked up. Earlier files are possibly related
         //       to libraries/frameworks that we rely upon.
         foreach (array_reverse(get_included_files()) as $includedFileName) {
+            if (! is_readable($includedFileName)) {
```

Comment:
Should not be better to use existing FileChecker?

Reply:
Good call!

Reply:
Actually, `FileChecker` operates with exceptions: not sure how heavy it could be, but it's probably not the hot path...